File: 200_LeetCode_Questions/Python/D11P3_LeetCode_33.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        def bsearch(nums, target, lo, hi):
            if lo == hi:
                return lo if nums[lo] == target else -1

            mid = (lo + hi) // 2
            if nums[mid] == target:
                return mid
            
            if (nums[hi] > nums[mid] and (target >= nums[mid] and target <= nums[hi])) or (nums[hi] < nums[mid] and (target <= nums[hi] or target >= nums[mid])):
                return bsearch(nums, target, mid + 1, hi)
            else:
                return bsearch(nums, target, lo, mid)
            
        return bsearch(nums, target, 0, len(nums) - 1)

File: 200_LeetCode_Questions/Java/D11P3_LeetCode_33.java
# Java code coming soon
class Solution {
    public int search(int[] nums, int target) {
         //If array is empty, return -1
        if(nums.length == 0) {
           return -1;
        }
        //Traverse an array
        for(int i = 0; i < nums.length; i++) {
 
            //If element is found, then return it's index
            if(nums[i] == target) {
                 return i;
            }
         }
         return -1;
    }
}
